operator/pkg/reconciliation/handler.go

```diff
@@ -192,9 +194,12 @@ func (rc *ReconciliationContext) isValid(dc *api.CassandraDatacenter) error {
 
 // NewReconciler returns a new reconcile.Reconciler
 func NewReconciler(mgr manager.Manager) reconcile.Reconciler {
+	client := mgr.GetClient()
+	dynamicWatches := dynamicwatch.NewDynamicSecretWatches(client)
```

Comment:
This instance needs to be available both during reconciliation _and_ when we set up our watches. Consequently, we put it on the `ReconcileCassandraDatacenter` and will pass along the reference to the `ReconciliationContext` when we create it.